frontend/src/scenes/data-warehouse/editor/output-pane-tabs/queryEndpointLogic.tsx

```diff
@@ -0,0 +1,118 @@
+import { actions, connect, kea, listeners, path, props, reducers } from 'kea'
+import { router } from 'kea-router'
+
+import api from 'lib/api'
+import { lemonToast } from 'lib/lemon-ui/LemonToast/LemonToast'
+import { EmbeddedTab } from 'scenes/embedded-analytics/common'
+import { queryEndpointsLogic } from 'scenes/embedded-analytics/query-endpoints/queryEndpointsLogic'
+import { urls } from 'scenes/urls'
+
+import { NamedQueryRequest, NodeKind } from '~/queries/schema/schema-general'
+
+import type { queryEndpointLogicType } from './queryEndpointLogicType'
+
+export type CodeExampleTab = 'terminal' | 'python' | 'nodejs'
+
+export interface QueryEndpointLogicProps {}
+
+export const queryEndpointLogic = kea<queryEndpointLogicType>([
+    path(['data-warehouse', 'editor', 'sidebar', 'queryEndpointLogic']),
+    props({} as QueryEndpointLogicProps),
+
```

Comment:
You'll probably want to key this by `tabId`, like I did here: https://github.com/PostHog/posthog/pull/38315 .